File: code_challenges/03_reverseCase.js
/*
Write a function that takes in a string, reverses the 'casing' of that string.


```
reverseCase('HelloWorld');
--> 'hELLOwORLD';

reverseCase('DOG cat');
--> 'dog CAT';
```
*/

const reverseCase = str => str
  .split('').reduce(
    (acc, c) =>
      c.toLowerCase() === c
        ? acc.concat(c.toUpperCase())
        : acc.concat(c.toLowerCase())
    , [])
  .join('')

// Old solution:
//const reverseCase = str => {
//  return str.split('').reduce((acc, l) => {
//    if(l.toLowerCase() === l) return acc.concat(l.toUpperCase())
//    else return acc.concat(l.toLowerCase())
//  }, [])
//  .join('')
//}

console.log(reverseCase('hItHeRe'))

File: code_challenges/clojure_solutions/03-reverse-case.clj
;; Write a function that takes in a string, reverses the 'casing' of that string.

;; ```
;; reverseCase('HelloWorld');
;; --> 'hELLOwORLD';

;; reverseCase('DOG cat');
;; --> 'dog CAT';
;; ```

(ns lambda-school
  (:require [clojure.string :as str]))

;; BROKEN! 01/28/18
(defn reverse-case [s]
  (str/replace s #"[a-z]" #(str (str/upper-case %1)))
  )

(reverse-case "DOG cat") ;; => dog CAT
